client/client.go

```diff
@@ -954,29 +976,30 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 
 	// if this is a root role, generate a root cert for the public key
 	if role == data.CanonicalRootRole {
-		privKey, _, err := r.CryptoService.GetPrivateKey(pubKey.ID())
-		if err != nil {
-			return err
-		}
-		pubKey, err = rootCertKey(r.gun, privKey)
-		if err != nil {
-			return err
+		for i, pubKey := range pubKeyList {
+			privKey, _, err := r.CryptoService.GetPrivateKey(pubKey.ID())
```

Comment:
I think `r.CryptoService.GetPrivateKey` returns a key, the role string, and the error?  So I think we can just use the second argument.  :)